src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -79,39 +76,35 @@ def initialize_jobs(self):
         for job_id, job_state in job_states.items():
             job = Job.from_state(job_state)
             status = job_state.get("status")
-            refresh_state = 1 if status not in TERMINAL_STATES else 0
+            refresh_state = 1 if status not in TERMINAL_STATUSES else 0
             self._running_jobs[job_id] = {
                 "refresh": refresh_state,
                 "job": job,
             }
-            # if the job is in a terminal state, add it to the _completed_job_states index
-            if not refresh_state:
-                revised_state = self._construct_job_status(
-                    self.get_job(job_id), job_state
-                )
-                self._completed_job_states[job_id] = revised_state
 
-    def _create_jobs(self, job_ids):
+    def _create_jobs(self, job_ids, job_states=None):
```

Comment:
Sometimes job_states already known. Is check_jobs even expensive?

Reply:
well, it's a call to ee2 (vs getting cached data), so it will be slower, and you can imagine a situation where ee2 does down, so it'd be useful to have that saved data.

Reply:
Agreed with @ialarmedalien. I'd avoid any EE2 calls if you can. You don't need to go super out of your way, though. It also depends on when `_create_jobs` gets called.

From elsewhere, it looks like `_create_jobs` only gets called during `retry_jobs`, where you don't have any job states, just the new retried ids. I'd ditch the `job_states` kwarg all together and just fetch the new state here.

This seems to have similar functionality to `initialize_jobs`, So maybe that could call this? Or both of these could call the same Job construction loop.

This also seems to return the new states. Could you make a note of that in the docstring and typing?